klog.go

```diff
@@ -1265,6 +1265,54 @@ func Infof(format string, args ...interface{}) {
 	logging.printf(infoLog, logging.logr, format, args...)
 }
 
+// InfoS structured logs to the INFO log.
+func InfoS(msg string, keysAndValues ...interface{}) {
+	b := &bytes.Buffer{}
+	kvMap, _ := kvList2Map(keysAndValues)
+	if kvMap == nil {
+		return
+	}
+
+	b.WriteString(msg)
+
+	for key, val := range kvMap {
+		appendKeyValue(b, key, val)
+	}
+
+	logging.println(infoLog, logging.logr, b)
```

Comment:
One of the goals of `InfoS` methods is correctly passing keysAndValues into logging.logr interface. Please look into internals of logging.println so that:
* If logging.logr != nil, then we call logging.logr.Info(msg, keysAndValues)
* otherwise, we use the format that you implemented

Reply:
add if statement